cluster-controller/src/main/java/io/strimzi/controller/cluster/Reconciliation.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * <p>Represents an attempt synchronize the state of some K8S resources (an "assembly") in a single namespace with a
+ * desired state, expressed as a ConfigMap.</p>
+ *
+ * <p>Each instance has a unique id and a trigger (description of the event which initiated the reconciliation),
+ * which are used to provide consistent context for logging.</p>
+ */
+public class Reconciliation {
+
+    private static final AtomicInteger IDS = new AtomicInteger();
+
+    private final String trigger;
+    private final String type;
+    private final String namespace;
+    private final String assemblyName;
+    private final int id;
+
+    public Reconciliation(String trigger, String type, String namespace, String assemblyName) {
```

Comment:
Could it be more understandable having an enum for the trigger ? watcher, timer and so on ?